api/csi_test.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCSIVolumes_CRUD(t *testing.T) {
+	t.Parallel()
+	c, s, root := makeACLClient(t, nil, nil)
+	defer s.Stop()
+	v := c.CSIVolumes()
+
+	// Successful empty result
+	vols, qm, err := v.List(nil)
+	assert.NoError(t, err)
+	assert.NotEqual(t, 0, qm.LastIndex)
+	assert.Equal(t, 0, len(vols))
+
+	// Authorized QueryOpts. Use the root token to just bypass ACL details
+	opts := &QueryOptions{
+		Region:    "global",
+		Namespace: "default",
+		AuthToken: root.SecretID,
+	}
+
+	wpts := &WriteOptions{
+		Region:    "global",
+		Namespace: "default",
+		AuthToken: root.SecretID,
+	}
+
+	// Register a volume
+	v.Register(&CSIVolume{
+		ID:             "DEADBEEF-63C7-407F-AE82-C99FBEF78FEB",
```

Comment:
Nitpicky / belt-and-suspenders: if we use a string literal here rather than generating the UUID uniquely per test, it's a lot easier for someone to come along later and pollute test state in a way that'll break `t.Parallel()` by copy-and-pasting-and-editing this test.

Reply:
That's a good point, I hadn't considered it...